app/services/sessionManager.js

```diff
@@ -1,14 +1,43 @@
 const jwt = require('jwt-simple'),
+  moment = require('moment'),
+  bcrypt = require('bcrypt'),
   config = require('./../../config');
 
 const SECRET = config.common.session.secret;
+const maxUsefulDays = parseInt(config.common.session.maxUsefulDays) || 30;
+const expirationDate = parseInt(config.common.session.expirationDate) || 2;
+const saltRounds = parseInt(config.common.bcrypt.saltRounds) || 10;
 
 exports.HEADER_NAME = config.common.session.header_name;
 
+const getMaximumUsefulDate = () => {
+  return moment().add(maxUsefulDays, 'days');
+};
+
+const getExpirationDate = () => {
+  return moment().add(expirationDate, 'days');
+};
+
 exports.encode = (toEncode) => {
   return jwt.encode(toEncode, SECRET);
 };
 
 exports.decode = (toDecode) => {
   return jwt.decode(toDecode, SECRET);
 };
+
+exports.generateTokenAccess = (user) => {
+  return bcrypt.genSalt(saltRounds).then((renewId) => {
+    return {
+      token: exports.encode({
+        authenticationCode: user.authenticationCode,
+        maximumUsefulDate: getMaximumUsefulDate(),
+        expirationDate: getExpirationDate(),
+        id: user.id,
+        renewId
+      }),
+      renewId,
+      expirationDate: getExpirationDate()
```

Comment:
We should call only one time the `getExpirationDate()` at one end to avoid the cost of calculating it two times and in another end so we send the exact same date in the two places (inside de jwt and in this other param).